aspnetcore/fundamentals/configuration/options.md

```diff
@@ -0,0 +1,256 @@
+
+In the following code, a third `IConfigureOptions<TOptions>` service is added to the service container. It binds `MySubOptions` to the section `subsection` of the *appsettings.json* file (*Startup.cs*):
+
+[!code-csharp[Main](options/sample/Startup.cs?name=snippet_Example3)]
+
+The `GetSection` extension method requires the [Microsoft.Extensions.Options.ConfigurationExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Options.ConfigurationExtensions/) NuGet package. If the app already uses the [Microsoft.AspNetCore.All](https://www.nuget.org/packages/Microsoft.AspNetCore.All/) metapackage, the package is automatically included.
+
+The sample's *appsettings.json* file defines a `subsection` member with keys for `suboption1` and `suboption2`:
+
+[!code-json[Main](options/sample/appsettings.json?highlight=4-7)]
+
+The `MySubOptions` class defines properties, `SubOption1` and `SubOption2`, to hold the sub-option values (*Models/MySubOptions.cs*):
+
+[!code-csharp[Main](options/sample/Models/MySubOptions.cs?name=snippet1)]
+
+The page model's `OnGet` method returns a string with the sub-option values (*Pages/Index.cshtml.cs*):
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?range=11)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet2&highlight=4,10)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet_Example3)]
+
+When the app is run, the `OnGet` method returns a string showing the sub-option class values:
+
+```html
+subOption1 = subvalue1_from_json, subOption2 = 200
+```
+
+## Options provided by a view model or with direct view injection
+
+Options provided by a view model or with direct view injection is demonstrated as Example &num;4 in the [sample app](https://github.com/aspnet/docs/tree/master/aspnetcore/fundamentals/configuration/options/sample).
+
+Options can also be supplied in a view model or by injecting `IOptions<TOptions>` directly into a view (*Pages/Index.cshtml.cs*):
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?range=9)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet2&highlight=2,8)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet_Example4)]
+
+For direct injection, inject `IOptions<MyOptions>` with an `@inject` directive:
+
+[!code-cshtml[Main](options/sample/Pages/Index.cshtml?range=1-10&highlight=5)]
+
+When the app is run, the option values are shown in the rendered page:
+
+![Options values Option1: value1_from_json and Option2: -1 are loaded from the model and by injection into the view.](options/_static/view.png)
+
+## Reload configuration data with IOptionsSnapshot
+
+Reload configuration data with `IOptionsSnapshot` as demonstrated in Example &num;5 in the [sample app](https://github.com/aspnet/docs/tree/master/aspnetcore/fundamentals/configuration/options/sample).
```

Comment:
as demonstrated in
did you mean **is** demo'd?

Reply:
It was to make "Reload" at the start of the sentence the verb (i.e., "[You] Reload ... as demonstrated ..."). I'm changing that now to make "is" the verb here to clean it up.